hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/PipelinePlacementPolicy.java

```diff
@@ -215,8 +216,9 @@ private boolean multipleRacksAvailable(List<DatanodeDetails> dns) {
    */
   @Override
   protected List<DatanodeDetails> chooseDatanodesInternal(
-      List<DatanodeDetails> excludedNodes, List<DatanodeDetails> favoredNodes,
-      int nodesRequired, long metadataSizeRequired, long dataSizeRequired)
+          List<DatanodeDetails> usedNodes, List<DatanodeDetails> excludedNodes,
```

Comment:
UsedNodes isn't used anywhere in this class, correct? If so, maybe add a comment to the Java doc for that parameter to highlight that.

I also wonder if we should do anything if someone passes something non-empty for used nodes here? Should we merge them into excludedNodes? Or throw an exception of some sort to hightlight the passed value will not actually be used?

Reply:
Currently this ticket is making changes for Rack scatter policy only.